### PR TITLE
Feature/search by rating

### PR DESCRIPTION
**Pull request by chadrick-d-dev**
## Description

- We created a dropdown button that a user can use to display movies within specified rating ranges.
- When no movies within a selected range are available a message will appear in MoviesContainer
- We refactored test that were affected by the creation of the new feature
- We wrote tests for the new feature

## Motivation and Context

- We needed to create an additional feature with testing per Iteration 5

## How Has This Been Tested?

- Unit and integration testing
- Refactored testing and wrote new tests within MoviesContainer, Wrote unit tests for RatingSearch, wrote integration tests in App test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Tickets:
#40 #41 #42

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -23,8 +25,16 @@ class App extends Component {
...
 
-  returnToHome = () => {
-    this.setState({ movieDetails: null })
+  getMoviesByRating = (ratingMinMax) => {
+    const range = ratingMinMax.split(","); 
+    const moviesInRange = this.state.movies.filter(movie => {
+      return movie.average_rating >= +range[0] && movie.average_rating <= +range[1]
+    })
+    if (moviesInRange.length === 0) {
+      this.setState({moviesByRating: ["NaN"]})
```
Comment (saraho1123) [inline src/App/App.js:34]: I like that this "NaN" worked. I wonder if there is a better way to do this. Let me know if you find a better one!

File: src/App/App.test.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import { render, screen, fireEvent, waitFor } from '@testing-library/react';
-import userEvent from '@testing-library/user-event'
+import userEvent from '@testing-library/user-event';
```
Comment (saraho1123) [inline src/App/App.test.js:2]: Did we ever end up using userEvent?

File: src/MovieCard/MovieCard.css [hunk 3]
```diff
@@ -4,15 +4,18 @@
   flex-direction: column;
   text-align: center;
   width: 20vw;
+  min-width: 244px;
```
Comment (saraho1123) [inline src/MovieCard/MovieCard.css:7]: I love the min-width, min-height for making pages responsive! It is simple and straightforward!

File: src/MoviesContainer/MoviesContainer.js [hunk 4]
```diff
@@ -2,24 +2,48 @@ import React from 'react';
...
-                id={movie.id}
-                key={movie.id}
-                image={movie.poster_path}
-                backdrop={movie.backdrop_path}
-                title={movie.title}
-                rating={movie.average_rating}
-            />
-        )
+const MoviesContainer = ({ movies, moviesByRating }) => {
+  if (moviesByRating[0] === "NaN") {
```
Comment (saraho1123) [inline src/MoviesContainer/MoviesContainer.js:6]: I like how our if statements work. I think this is the best way to do this. At least at my experience so far, it works!

File: src/MoviesContainer/MoviesContainer.test.js [hunk 5]
```diff
@@ -51,4 +53,59 @@ describe('MovieContainer', () => {
     expect(mulanImage).toBeInTheDocument();
     expect(rogueImage).toBeInTheDocument();
   })
+
```
Comment (saraho1123) [inline src/MoviesContainer/MoviesContainer.test.js:56]: I'm glad we got more testing practice! I feel like I can write tests now. I'm not sure they are exactly correct or if they are always in the right place, but at least I am understanding them a bit now!

File: src/RatingSearch/RatingSearch.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import './RatingSearch.css';
+import { Dropdown, DropdownButton } from 'react-bootstrap';
+
+const RatingSearch = ({ getMoviesByRating }) => {
+  return(
+    <DropdownButton id="dropdown-basic-button" title="Search Movies By Rating" className="ratings-dropdown" >
+      <Dropdown.Item onClick={(event) => getMoviesByRating(event.target.name)} name={[1, 10]} href="/">All Movies</Dropdown.Item>
```
Comment (saraho1123) [inline src/RatingSearch/RatingSearch.js:8]: Dropdowns simplify sad-path testing. lol!

File: src/SingleMovie/SingleMovie.test.js [hunk 7]
```diff
@@ -10,7 +10,7 @@ jest.mock('../apiCalls.js');
 
 
 describe('SingleMovie', () => {
-  it('should render details for a single movie', async () => {
+  beforeEach(() => {
```
Comment (saraho1123) [inline src/SingleMovie/SingleMovie.test.js:13]: Thanks for getting that beforeEach figured out!!

File: src/App/App.test.js (hunk 2 above)
Comment (chadrick-d-dev) [reply src/App/App.test.js:2]: Yes, we use it in one of the tests in this file.

File: src/apiCalls.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ export const getAllMoviesData = () => {
 }
 
 export const getSingleMovieData = (id) => {
-  const apiLink = 'https://rancid-tomatillas.herokuapp.com/api/v2//movies/'
+  const apiLink = 'https://rancid-tomatillos.herokuapp.com/api/v2//movies/'
```
Comment (saraho1123) [inline src/apiCalls.js:7]: Thanks for catching that from when we were running tests!

File: src/MovieCard/MovieCard.css (hunk 3 above)
Comment (chadrick-d-dev) [reply src/MovieCard/MovieCard.css:7]: I like it a lot too!

File: src/index.js [hunk 9]
```diff
@@ -4,6 +4,7 @@ import './index.css';
 import App from './App/App';
 import reportWebVitals from './reportWebVitals';
 import { BrowserRouter } from 'react-router-dom';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (saraho1123) [inline src/index.js:7]: bootstrap is pretty cool!

File: src/MoviesContainer/MoviesContainer.js (hunk 4 above)
Comment (chadrick-d-dev) [reply src/MoviesContainer/MoviesContainer.js:6]: I agree!

**Review by saraho1123 — APPROVED** [review]
we did it Chadrick! Nice work!

File: src/MoviesContainer/MoviesContainer.test.js (hunk 5 above)
Comment (chadrick-d-dev) [reply src/MoviesContainer/MoviesContainer.test.js:56]: I found it to be very rewarding!

File: src/RatingSearch/RatingSearch.js (hunk 6 above)
Comment (chadrick-d-dev) [reply src/RatingSearch/RatingSearch.js:8]: They really do!

File: src/index.js (hunk 9 above)
Comment (chadrick-d-dev) [reply src/index.js:7]: So cool!

**Pull request merged**